src/platform/silabs/wifi/wf200/WifiInterface.cpp

```diff
@@ -305,6 +303,33 @@ CHIP_ERROR GetMacAddress(sl_wfx_interface_t interface, MutableByteSpan & address
     return CopySpanToMutableSpan(byteSpan, address);
 }
 
+CHIP_ERROR StartNetworkScan(chip::ByteSpan ssid, ScanCallback callback)
+{
+    VerifyOrReturnError(callback != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+
+    // SSID Max Length that is supported by the Wi-Fi SDK is 32
+    VerifyOrReturnError(ssid.size() <= WFX_MAX_SSID_LENGTH, CHIP_ERROR_INVALID_STRING_LENGTH);
+
+    if (ssid.empty())
+    {
+        scan_ssid_length = 0;
+        scan_ssid        = nullptr;
+    }
+    else
+    {
+        scan_ssid_length = ssid.size();
+        scan_ssid        = reinterpret_cast<uint8_t *>(chip::Platform::MemoryAlloc(scan_ssid_length));
+        VerifyOrReturnError(scan_ssid != nullptr, CHIP_ERROR_NO_MEMORY);
```

Comment:
At first sight, it is not obvious that `scan_ssid` is a global variable.

Reply:
Agreed; wll keep that clean up for a follow up when we start refactoring the interface implementations.